### PR TITLE
#169258608 User get all trips made between specific dates

### PR DESCRIPTION
**Pull request by william000000**
#### What does this PR do?
Users should get stats of trips made in the last X timeframe
#### Description of Task to be completed?
- Design the components
- Consume GET Request API for this functionality
#### How should this be manually tested?
- `git clone https://github.com/andela/team-odd-bn-frontend`
- `git checkout ft-get-stats-trips-in-the-last-timeframe-169258608
- `npm run install` to install all dependencies
- `npm run dev` to start the development server
- `Login`
- Navigate to `/trips/stats` route
#### What are the relevant pivotal tracker stories?
[#169258608](https://www.pivotaltracker.com/story/show/169258608)
#### Screenshots (if appropriate)

image

### CODE REVIEW

File: src/containers/trips/SearchStats.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    if (this.props.profileError.error
+  || this.props.profileError.message === 'You have provided an invalid token') {
+      return this.props.history.push('/signin');
+    }
+  }
+
+  render() {
+    const { getStatsData } = this.props.stateObject.trips.getStats;
+    console.log('kkkkkkk', getStatsData);
```
Comment (victkarangwa) [inline src/containers/trips/SearchStats.js]: Could you please remove unnecessary logging?

File: src/redux/actions/tripsActions/statsTripActions.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    const { data } = await error.response;
+    await notificationError(`${data.message}`);
+    return dispatch(getStatsError(`${data.message}`));
+  }
+};
+
+
+export const handleInputsAction = (data) => {
+  console.log('dattttt==', data);
+
```
Comment (victorkarangwa4) [inline src/redux/actions/tripsActions/statsTripActions.js]: I think this comment is unnecessary.

File: src/views/trips/SearchTripStats.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import CommonTable from './CommonTable';
+import Action from '../../components/Action';
+import Status from '../../components/Status';
+import { searchNavs, searchTableHeads } from '../../constants/trips';
+
+const RequestsView = ({ data }) => {
+  console.log('datta====>', data);
```
Comment (victkarangwa) [inline src/views/trips/SearchTripStats.js]: Unnecessary console logging

File: src/views/trips/SearchTripStats.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  const entities = data ? data.map((request) => [
+    { className: 'tripId', attribute: request.id, key: 'tripId' },
+    { className: 'tripType', attribute: request.originId, key: 'tripType' },
+    { className: 'createdAt', attribute: request.destinationId, key: 'createdAt' },
+    { className: 'createdAt', attribute: request.createdAt.slice(0, 10), key: 'createdAt' },
+    { className: 'action', attribute: <Action background="#34c6f3" url={`/requests/${request.id}`} action="View" />, key: 'action' },
+  ]) : [];
+  console.log('datta====>', entities);
+
```
Comment (victkarangwa) [inline src/views/trips/SearchTripStats.js]: I think this is unnecessary too.

**Review by victkarangwa — CHANGES_REQUESTED** [review]
So far so good @william000000 
However, you can get rid of some unnecessary `console.log`

File: src/components/StatsNav.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        </button>
+      </div>
+
+    </div>
+  </div>
+
+);
+
+const mapStateToProps = (state) => {
+  console.log('statet----', state.trips.getStats.getStatsInput);
```
Comment (hezronkimutai) [inline src/components/StatsNav.js]: Make sure you remove the console.

**Review by hezronkimutai — APPROVED** [review]

**Review by victkarangwa — APPROVED** [review]

**Review by shemaeric — APPROVED** [review]
Good work @william000000

**Comment by NiyongaboEric** [comment]
Good job @william000000.  I like the IPAD responsiveness.

image

**Review by NiyongaboEric — APPROVED** [review]

**Pull request merged**